gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -243,6 +244,24 @@ def sign_request(request)
         )
       end
 
+      # Signs a event and returns signature headers and prior signature
```

Comment:
Let's flesh out this documentation with documentation of parameters and the return value. This method is doing a lot of stuff (for one example, explaining how we're changing the encoding at the end with the unpacking).